workers/unity/Assets/Fps/Scripts/SetupLogic/ConnectionController.cs

```diff
@@ -12,93 +13,87 @@ public class ConnectionController : MonoBehaviour
+        private void OnEnable()
         {
-            if (obj.StatusCode == StatusCode.Success)
-            {
-                canvasCameraObj.SetActive(false);
-                screenUIController.ConnectScreen.SetActive(false);
-                screenUIController.InGameHud.SetActive(true);
-            }
-            else
+            if (responseHandler != null)
             {
-                connectButton.SetTrigger("FailedToSpawn");
+                responseHandler.OnCreatePlayerResponse += OnCreatePlayerResponse;
             }
         }
 
-        private void Update()
+        private void OnCreatePlayerResponse(PlayerCreator.CreatePlayer.ReceivedResponse obj)
         {
-            if (!connectButton.isActiveAndEnabled)
+            if (obj.StatusCode == StatusCode.Success)
             {
-                return;
+                ConnectionStateReporter.SetState(ConnectionStateReporter.State.Spawned);
             }
-
-            if (clientWorkerConnector != null && clientWorkerConnector.Worker != null)
+            else
             {
-                // Worker successfully connected
-                if (connectButton.GetCurrentAnimatorStateInfo(0).IsName("ConnectingState"))
-                {
-                    connectButton.SetTrigger("Ready");
-                }
+                ConnectionStateReporter.SetState(ConnectionStateReporter.State.SpawningFailed,
+                    obj.StatusCode.ToString());
             }
         }
 
-        public void OnFailedToConnect()
+        public void OnFailedToConnect(string errorMessage)
         {
-            // Worker failed to connect
-            connectButton.SetTrigger("FailedToConnect");
+            ConnectionStateReporter.SetState(ConnectionStateReporter.State.ConnectionFailed, errorMessage);
         }
 
-
-        private void SpawnPlayer()
+        public void OnDisconnected()
         {
-            var request = new CreatePlayerRequestType(new Vector3f { X = 0, Y = 0, Z = 0 });
-            commandSender.SendCreatePlayerRequest(new EntityId(1), request);
+            ConnectionStateReporter.SetState(ConnectionStateReporter.State.WorkerDisconnected);
         }
 
         public void ConnectAction()
         {
-            if (connectButton.GetCurrentAnimatorStateInfo(0).IsName("ReadyState"))
-            {
-                connectButton.SetTrigger("Connecting");
-                SpawnPlayer();
-            }
-            else if (connectButton.GetCurrentAnimatorStateInfo(0).IsName("FailedToSpawn"))
-            {
-                connectButton.SetTrigger("Retry");
-                SpawnPlayer();
-            }
-            else if (connectButton.GetCurrentAnimatorStateInfo(0).IsName("FailedToConnect")
-                || connectButton.GetCurrentAnimatorStateInfo(0).IsName("WorkerDisconnected"))
+            ClientWorkerHandler.CreateClient();
+        }
+
+        public void DisconnectAction()
```

Comment:
Why are these functions called `SomethingAction` instead of just `Something`? Seems redundant